schemas/x509.yml

```diff
@@ -0,0 +1,171 @@
+---
+- name: x509
+  title: x509 Certificate
+  group: 2
+  short_description: These fields contain x509 certificate metadata.
+  description: >
+    This implements the common core fields for x509 certificates. This information is likely logged with TLS sessions,
+    digital signatures found in executable binaries, S/MIME information in email bodies, or analysis of files on disk.
+    When only a single certificate is logged in an event, it should be nested under `file`. When hashes of the DER-encoded
+    certificate are available, the `hash` data set should be populated as well (e.g. `file.hash.sha256`). For events that
+    contain certificate information for both sides of the connection, the x509 object could be nested under the respective
+    side of the connection information (e.g. `tls.server.x509`).
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - file
+      - tls.client
+      - tls.server
+  fields:
+    - name: version_number
+      level: extended
+      type: keyword
+      description: Version of x509 format.
+      example: 3
+
+    - name: serial_number
+      level: extended
+      type: keyword
+      short_description: Unique serial number issued by the certificate authority.
+      description: >
+        Unique serial number issued by the certificate authority. For consistency, if this value is alphanumeric, it should be
+        formatted without colons and uppercase characters.
+      example: 55FBB9C7DEBF09809D12CCAA
+
+    - name: issuer.distinguished_name
+      level: extended
+      type: keyword
+      description: Distinguished name (DN) of issuing certificate authority.
+      example: C=US, O=DigiCert Inc, OU=www.digicert.com, CN=DigiCert SHA2 High Assurance Server CA
+
+    - name: issuer.common_name
+      level: extended
+      type: keyword
+      description: Common name (CN) of issuing certificate authority.
+      example: DigiCert SHA2 High Assurance Server CA
+
+    - name: issuer.organizational_unit
+      level: extended
+      type: keyword
+      normalize:
+        - array
+      description: List of organizational units (OU) of issuing certificate authority.
+      example: www.digicert.com
+
+    - name: issuer.organization
```

Comment:
Seems that this one could also potentially be an array.

In Go standard library, all of `country`, `organization`, `organizational_unit`, `locality` and `province` are arrays, but it's not that we have to tailor this to match what Go does.

Reply:
That's a good point, actually. I'm not worried about matching what Golang does per se.

If they have them all as arrays, it's probably a sign that all of these attributes can be a list of values. I've heard that Googles knows a bit about internet protocols & stuff 😂

Reply:
But it's worth double-checking the protocol definition that ones like "locality", "province" and "country" are actually arrays.

It could also be that it was just simpler to do it this way in Golang for other reasons, and that it's not needed. Not sure.

Reply:
After looking at RFC after RFC, ASN.1 files, etc...just gave it a go in OpenSSL. Today I Learned

```
openssl req -new -out myserver.csr \
>   -newkey rsa:2048 -sha1 -nodes -keyout myserver.key  \
>   -subj "/C=US/C=CA/ST=Texas/ST=Ontario/O=Elastic/O=Elastic Engineering/OU=SIEM/OU=ECS/OU=Intelligence and Analytics/CN=Derek/CN=Mat"
```

Then `openssl req -in myserver.csr -noout -text`. Apparently x.509 is the wild west. Do whatever you want.

```
Certificate Request:
    Data:
        Version: 0 (0x0)
        Subject: C=US, C=CA, ST=Texas, ST=Qu\xC3\x83\xC2\xA9bec, O=Elastic, O=Elastic Engineering, OU=SIEM, OU=ECS, OU=Intelligence and Analytics, CN=Derek, CN=Mat
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                Public-Key: (2048 bit)
                Modulus:
...
```

I'll make these all lists.